diff --git a/src/logging.js b/src/logging.js
index a4dfec3..1cddf7b 100644
--- a/src/logging.js
+++ b/src/logging.js
@@ -37,6 +37,9 @@ module.exports = {
         }
         this.log("Minimisation finished; final version is at %s (%d bytes)", file, size);
     },
+    logLocDone: function (file) {
+        this.log("Source-code location information for final file is at %s", file);
+    },
     increaseIndentation: function () {
         indentation.push("  ")
     },
@@ -46,4 +49,4 @@ module.exports = {
     getIndentation: function () {
         return formatIndentation();
     }
-};
\ No newline at end of file
+};
